#167727725 send email verification on signup

kola-1 opened the pull request:
#### What this PR does
This PR will send email verification to the user on signup

#### Tasks completed
- install sendgrid
- send email to user on signup with sendgrid

#### How this can be manually tested
- Clone this branch
- Change directory into it
- Run `npm install` in your terminal to install all the dependencies
- Add all the environment variables as specified in the `.env.config` file
- Run `npm run start:dev` in your terminal to start the development server
- Test with postman or go to this url to test with swagger `http://localhost:3000/api/v1/docs`
- Signup with your email address at this endpoint `POST api/v1/user/signup`
- In your email which you signed up with you will get an email verification message

#### What are the relevant pivotal tracker stories?
[#167727725](https://www.pivotaltracker.com/story/show/167727725)

ladaposamuel commented on src/config/verifyEmailMessage.js:
```diff
@@ -0,0 +1,323 @@
+  return { text, html };
+};
+
+export default verifyEmailMessage;
```
I don't think the email template should be this large, and also it will be awesome to have a master layout for emails like so:

```javascript

export const EmailTemplate = (title,message,data) => `
<div style="background: #fff; width: 100%; height: 100%; padding: 3rem 0; font-family: 'Georgia'">
  <div style="width: 50vw; background: #FFF; display: block; margin: 0 auto; padding: 0; border: 1px solid #a00b40;">
    <div style=" background: #a00b40;">
      <h2 style="color: white; text-align: center; line-height: 10vh; background: #a00b40; margin: 0;">PropertyPro-Lite</h2>
    </div>
    <div style="text-align: center; padding: 0.5rem;">
      <p style="font-size: 1rem;">${title},</p>
      <p style="font-size: 1rem;">${message}</p>
      <p style="padding: 0.3rem;"></p>
      <hr>
      <center>${data.newPassword}</center>
      <hr>
      <p style="padding: 0.3rem;"></p>
      <p style="font-size: 1rem;">Bearfoor Nomad © ${data.year}</p>
    </div>
  </div>
</div>`;

```

ladaposamuel replied:
This is a sample of how the email will look like.
![Screenshot 2019-08-22 at 10 33 24 AM](https://user-images.githubusercontent.com/9302799/63503902-516b4a80-c4c8-11e9-8061-d8a542c7d3e4.png)

ladaposamuel commented on package.json:
```diff
@@ -13,6 +13,7 @@
     "lint:fix": "./node_modules/.bin/eslint --fix",
     "start": "set DEBUG=http:* && node ./dist/index.js",
     "start:dev": "set DEBUG=http:* && nodemon src/index.js --exec babel-node",
+    "send:mail": "set DEBUG=http:* && babel-node src/sendMail.js",
```
What does this script do?

kola-1 committed:
feature(email): send email verification on signup

- install sendgrid
- add verified column to user table
- add email verification route

[Finishes #167727725]

kola-1 commented on src/controllers/emailVerificationController.js:
```diff
@@ -0,0 +1,29 @@
+
+const verifyEmail = async (req, res) => {
+  try {
+    const { id } = await jwt.verify(req.params.token, process.env.SECRET_KEY);
```
extract line 15 to authHelper

jsbuddy commented on src/utils/createTemplate.js:
```diff
@@ -0,0 +1,19 @@
+  */
+const createTemplate = (template, link) => {
+  const { text } = template(link);
+  const { html } = template(link);
```
Is it possible to destructure `text` and `html` like so `const { text, html } = template(link)`

kola-1 replied:
Done. Thanks, that's actually better

ladaposamuel replied:
Yes

kola-1 commented on src/database/migrations/20190815205048-create-user.js:
```diff
@@ -36,6 +36,10 @@ module.exports = {
+        },
+        verified: {
+          allowNull: false,
+          type: Sequelize.BOOLEAN
```
allowNull should be true. defaultValue should be false

kola-1 commented on src/database/models/user.js:
```diff
@@ -37,6 +37,11 @@ export default (Sequelize, DataTypes) => {
       type: DataTypes.DATE
+    },
+    verified: {
+      allowNull: false,
```
allowNull should be true

kola-1 commented on src/database/migrations/20190815205048-create-user.js:
```diff
@@ -36,6 +36,10 @@ module.exports = {
         updatedAt: {
           allowNull: false,
```
allowNull should be true. defaultValue should be same as described below

kola-1 commented on src/database/models/user.js:
```diff
@@ -37,6 +37,11 @@ export default (Sequelize, DataTypes) => {
     updatedAt: {
       allowNull: false,
```
allowNull should be true. defaultValue should be sequelize.fn('now')

tejiri4 reviewed (changes requested):
Rebase with develop when you done

tejiri4 commented on src/database/models/user.js:
```diff
@@ -32,11 +32,18 @@ export default (Sequelize, DataTypes) => {
+      defaultValue: Sequelize.fn('NOW')
     },
     updatedAt: {
       allowNull: false,
```
change allowNull to true

kola-1 replied:
Done

kola-1 replied:
Done

tejiri4 commented on src/database/migrations/20190815205048-create-user.js:
```diff
@@ -31,11 +31,18 @@ module.exports = {
+          defaultValue: Sequelize.fn('NOW')
         },
         updatedAt: {
           allowNull: false,
```
change allow null to true

kola-1 replied:
Done

kola-1 commented:
> Rebase with develop when you done

Done

tejiri4 reviewed (approved):
Nice work LGTM
